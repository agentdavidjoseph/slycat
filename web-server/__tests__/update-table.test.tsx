import React from 'react';
import { mount } from "enzyme";
import ControlsButtonUpdateTable, {ControlsButtonUpdateTableProps, ControlsButtonUpdateTableState} from '../plugins/slycat-parameter-image/js/Components/ControlsButtonUpdateTable.jsx';
interface GlobalAny extends NodeJS.Global {
    fetch: any
}
function flushPromises(): any {
    return new Promise(resolve => setImmediate(resolve));
}
const globalAny:GlobalAny = global;
describe('When loading the update table modal',() =>{
    beforeEach(() => {
        globalAny.fetch.resetMocks()
    });
    test('We expect props on initial load', async () => {
        const properties: ControlsButtonUpdateTableProps = {
            button_style: 'btn-outline-dark',
            mid: '123',
            pid: '456'
        };
        const render = await mount(
            <ControlsButtonUpdateTable
                {...properties}
            />
        );
        const props = render.props() as ControlsButtonUpdateTableProps;
        expect(props).toMatchSnapshot();
    });

    test('We have proper state', async () => {
        const properties: ControlsButtonUpdateTableProps = {
            button_style: 'btn-outline-dark',
            mid: '123',
            pid: '456'
        };
        const render = await mount(
            <ControlsButtonUpdateTable
                {...properties}
            />
        );
        const state = render.state() as ControlsButtonUpdateTableState;
        expect(state).toMatchSnapshot();
    });

    test('We can clean up state', async () => {
        const properties: ControlsButtonUpdateTableProps = {
            button_style: 'btn-outline-dark',
            mid: '123',
            pid: '456'
        };
        const render = await mount(
            <ControlsButtonUpdateTable
                {...properties}
            />
        );
        const instance:any = render.instance() as any;
        jest.spyOn(instance, 'cleanup');
        const initialState = instance.state;
        await instance.cleanup();
        expect(instance.cleanup).toHaveBeenCalled();
        const cleanedState = instance.state;
        expect(initialState).toEqual(cleanedState);
    });
});