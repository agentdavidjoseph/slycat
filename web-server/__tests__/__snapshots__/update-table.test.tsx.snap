// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading the update table modal We expect props on initial load 1`] = `
Object {
  "button_style": "btn-outline-dark",
  "mid": "123",
  "pid": "456",
}
`;

exports[`When loading the update table modal We have proper state 1`] = `
Object {
  "currentColumns": Array [],
  "disabled": true,
  "failedColumnCheckMessage": Array [
    "This upload attempt has been rejected.
",
  ],
  "files": Array [
    File {},
  ],
  "hostname": "",
  "loadingData": false,
  "modalId": "varUpdateTableModal",
  "newColumns": Array [],
  "parserType": "slycat-csv-parser",
  "passedColumnCheck": true,
  "password": "",
  "progressBarHidden": true,
  "progressBarProgress": 0,
  "reauth": false,
  "selectedNameIndex": 0,
  "selectedOption": "local",
  "selected_path": "",
  "sessionExists": null,
  "title": "Update Table",
  "username": "",
  "visible_tab": "0",
}
`;
